File: src/Engine/Core/WeaponPoint.cpp

#include "WeaponPoint.h"

//public: UByte frame;

const int WeaponPoint::INVALID_WEAPON_POINT = -10000;

//--------------------
// -- CONSTRUCTORS --
//--------------------
WeaponPoint::WeaponPoint(int x, int y, UByte angle, bool inFront):
    Point16(x,y),angle(angle), inFront(inFront)
{}

WeaponPoint::WeaponPoint(DataInputStream& dis){ //throws IOException
    readData(dis);
}

WeaponPoint::WeaponPoint():Point16(),angle(0), inFront(false){}
WeaponPoint::WeaponPoint(const WeaponPoint& other):
    Point16(other),angle(other.angle), inFront(other.inFront)
{}


//---------------
// -- METHODS --
//---------------

void WeaponPoint::readData(DataInputStream& dis){ //throws IOException
    x = dis.readInt16();
    if (x!=INVALID_WEAPON_POINT){
        y = dis.readInt16();
        angle = (dis.readByte()*360)/256;
        inFront = dis.readBool();
    }
}


//public void writeData(DataOutputStream dos) throws IOException {
//    dos.writeShort(x);
//    dos.writeShort(y);
//    frame.writeData(dos);
//}

File: src/Editors/Editors Lib/src/characters/WeaponPoint.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package characters;

import file.IReadWrite;
import game.*;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import util.UByte;

/**
 *
 * @author Gil
 */
public class WeaponPoint implements IReadWrite{
    public static final int INVALID_WEAPON_POINT = -10000;
    //--------------
    // -- FIELDS --
    //--------------
    public int x;
    public int y;
    public UByte angle;
    public boolean inFront;
    
    //--------------------
    // -- CONSTRUCTORS --
    //--------------------
    public WeaponPoint(int x, int y, int angle, boolean inFront){
        this.x = x; this.y = y;
        this.angle = UByte.degrees(angle);
        this.inFront = inFront;
    }

    public WeaponPoint(DataInputStream dis) throws IOException {
        this.readData(dis);
    }
    public WeaponPoint(){
        x = 0; y = 0; angle = new UByte(); inFront = true;
    }
    public WeaponPoint(WeaponPoint other){
        this.x = other.x;
        this.y = other.y;
        this.angle = new UByte(other.angle.get());
        this.inFront = other.inFront;
    }
    

    //---------------
    // -- METHODS --
    //---------------
    
    public void writeData(DataOutputStream dos) throws IOException {
        dos.writeShort(x);
        if (x!=INVALID_WEAPON_POINT){
            dos.writeShort(y);
            angle.writeData(dos);
            dos.writeBoolean(inFront);
        }
    }

    public void readData(DataInputStream dis) throws IOException {
        x = dis.readShort();
        if (x!=INVALID_WEAPON_POINT){
            y = dis.readShort();
            angle = new UByte(dis);
            inFront = dis.readBoolean();
        }
    }
    
    
}
